gutterball/src/main/java/org/candlepin/gutterball/curator/ComplianceSnapshotCurator.java

```diff
@@ -1149,5 +1157,52 @@ private int compareDatesByDate(Date date1, Date date2) {
         return (year != 0 ? year : (month != 0 ? month : date));
     }
 
+    /**
+     * Deletes any compliances older than the specified duration in minutes.
+     *
+     * @param minutes
+     *  The number of minutes to use for cleaning up compliances
+     *
+     * @return
+     *  the number of compliance snapshots deleted during the cleanup operation
+     */
+    @Transactional
+    public int cleanupCompliances(int minutes) {
```

Comment:
Does making this transactional mean that if there are a thousand compliances and one will not delete, none will?